activerecord/lib/active_record/relation/query_methods.rb

```diff
@@ -2110,14 +2110,14 @@ def check_if_method_has_arguments!(method_name, args, message = nil)
       def process_select_args(fields)
         fields.flat_map do |field|
           if field.is_a?(Hash)
-            arel_columns_from_hash(field)
+            arel_columns_from_hash(field, for_select: true)
           else
             field
           end
         end
       end
 
-      def arel_columns_from_hash(fields)
+      def arel_columns_from_hash(fields, for_select: false)
```

Comment:
This is a bit hacky, see my note in the 'Additional information ' section for a possible alternative.